### PR TITLE
Rework connector expoerter architecture

### PR DESCRIPTION
**Pull request by junkerm**
[Trello Card]()

### CODE REVIEW

File: web/src/app/modules/actions/modules/test-export-button/components/test-export-button.component.ts [hunk 1]
```diff
@@ -36,24 +39,32 @@ export class TestExportButton {
...
             {
-                language: new LowerCasePipe().transform(this._lang),
-                exportDate: (new Date().getTime()).toString()
+                language: new LowerCasePipe().transform(this._lang)
             });
 
-        if (data === undefined) {
-            throw new Error('Could not load test specification skeleton for ' + this._lang);
-        }
+        if (data === null) {
```
Comment (sebeder) [inline web/src/app/modules/actions/modules/test-export-button/components/test-export-button.component.ts]: I'd add an undefined check as well. We use undefined everywhere and getting `null` is rather uncommon. Someone might change that in the future.

File: web/src/app/modules/views/side/modules/links-actions/services/additional-information.service.ts [hunk 2]
```diff
@@ -72,6 +75,14 @@ export class AdditionalInformationService {
         this.parents = await Promise.all(parentUrls.map(url => this.dataService.readElement(url)));
     }
 
+    private async loadExports(): Promise<void> {
+        if(Type.is(this.element,TestSpecification) || Type.is(this.element, TestProcedure)){
```
Comment (sebeder) [inline web/src/app/modules/views/side/modules/links-actions/services/additional-information.service.ts]: Please format correctly.

**Review by sebeder — CHANGES_REQUESTED** [review]
I have minor comments on the code. However, this one is too large to be reviewed well. We will need to try it out and talk about it a little bit.

File: web/src/app/modules/actions/modules/test-export-button/components/test-export-button.component.ts (hunk 1 above)
Comment (junkerm) [reply web/src/app/modules/actions/modules/test-export-button/components/test-export-button.component.ts]: added check

File: web/src/app/modules/views/side/modules/links-actions/services/additional-information.service.ts (hunk 2 above)
Comment (junkerm) [reply web/src/app/modules/views/side/modules/links-actions/services/additional-information.service.ts]: formatted correctly

File: web/src/app/modules/actions/modules/test-export-button/components/test-export-button.component.ts [hunk 3]
```diff
@@ -36,24 +39,32 @@ export class TestExportButton {
...
             {
-                language: new LowerCasePipe().transform(this._lang),
-                exportDate: (new Date().getTime()).toString()
+                language: new LowerCasePipe().transform(this._lang)
             });
 
-        if (data === undefined) {
-            throw new Error('Could not load test specification skeleton for ' + this._lang);
-        }
+        if (data === null || data===undefined) {
```
Comment (sebeder) [inline web/src/app/modules/actions/modules/test-export-button/components/test-export-button.component.ts]: Check your formatting ;-)

**Review by sebeder — CHANGES_REQUESTED** [review]

File: web/src/app/modules/actions/modules/test-export-button/components/test-export-button.component.ts (hunk 3 above)
Comment (junkerm) [reply web/src/app/modules/actions/modules/test-export-button/components/test-export-button.component.ts]: done

**Comment by sebeder** [comment]
I just tested the changes. They're fine for me once the pipeline passes.

**Review by sebeder — APPROVED** [review]

**Review by sebeder — APPROVED** [review]

**Pull request merged**